iam.tf

```diff
@@ -3,11 +3,43 @@ data "aws_caller_identity" "current" {
 
 resource "aws_iam_role" "lambda_execution_role" {
   assume_role_policy = var.lambda_assume_role
-
+  tags = var.tags
```

Comment:
In terms of tags, I would like the infrablocks modules to always include `Component`, `DeploymentIdentifier`, `DeploymentType` and `DeploymentLabel` and then the user provided tags are additional.

I've contemplated adding an option not to include default tags but haven't added this anywhere yet. I don't want the user to have to repeat the default tags when they only want to pass in one extra.

See https://github.com/infrablocks/terraform-aws-ecs-cluster for an example of what I mean. (https://github.com/infrablocks/terraform-aws-ecs-cluster/blob/master/locals.tf)